internal/creds.go

```diff
@@ -64,54 +65,63 @@ const (
 
 // credentialsFromJSON returns a google.Credentials based on the input.
 //
-// - A self-signed JWT auth flow will be executed if: the data file is a service
-//   account, no user are scopes provided, an audience is provided, a user
-//   specified endpoint is not provided, and credentials will not be
-//   impersonated.
+// - A self-signed JWT flow will be executed if the following conditions are
+// met:
+//   (1) Either the scope for self-signed JWT flow is enabled or audiences are
+//       explicitly provided by users.
+//   (2) No service account impersontation
 //
-// - Otherwise, executes a stanard OAuth 2.0 flow.
+// - Otherwise, executes standard OAuth 2.0 flow
+// More details: google.aip.dev/auth/4111
 func credentialsFromJSON(ctx context.Context, data []byte, ds *DialSettings) (*google.Credentials, error) {
+	// By default, a standard OAuth 2.0 token source is created
 	cred, err := google.CredentialsFromJSON(ctx, data, ds.GetScopes()...)
 	if err != nil {
 		return nil, err
 	}
-	// Standard OAuth 2.0 Flow
-	if len(data) == 0 ||
-		len(ds.Scopes) > 0 ||
-		(ds.DefaultAudience == "" && len(ds.Audiences) == 0) ||
-		ds.ImpersonationConfig != nil ||
-		ds.Endpoint != "" {
```

Comment:
Is this meant to go away? Endpoint now can work with self-signed /w scopes but it does not when aud is used.

Reply:
Right I don't think we need this anymore. When we use self-signed JWT there are two conditions:

1) Scope is enabled for self-signed JWT: in this case the scope will be used so custom endpoint will not an issue.
2) User provided audience: the endpoint is not an issue either since the user is responsible of the audience.

Reply:
Ack